pkg/kf/commands/services/create-service.go

```diff
@@ -49,19 +59,103 @@ func NewCreateServiceCommand(p *config.KfParams, client services.ClientInterface
 			if err != nil {
 				return err
 			}
+			paramBytes, err := json.Marshal(params)
+			if err != nil {
+				return err
+			}
+			rawParams := &runtime.RawExtension{
+				Raw: paramBytes,
+			}
 
-			instance, err := client.CreateService(
-				instanceName,
-				serviceName,
-				planName,
-				services.WithCreateServiceNamespace(p.Namespace),
-				services.WithCreateServiceParams(params))
+			clusterPlans, err := client.ServicecatalogV1beta1().
+				ClusterServicePlans().
+				List(metav1.ListOptions{})
 			if err != nil {
 				return err
 			}
 
-			describe.ServiceInstance(cmd.OutOrStdout(), instance)
-			return nil
+			for _, plan := range clusterPlans.Items {
+				if planName != plan.Spec.ExternalName {
+					continue
+				}
+				if serviceName != plan.Spec.ClusterServiceClassRef.Name {
+					continue
+				}
+				if broker != "" && broker != plan.Spec.ClusterServiceBrokerName {
+					continue
+				}
+
+				// plan found
+				created, err := client.ServicecatalogV1beta1().
+					ServiceInstances(p.Namespace).
+					Create(&servicecatalogv1beta1.ServiceInstance{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      instanceName,
+							Namespace: p.Namespace,
+						},
+						Spec: servicecatalogv1beta1.ServiceInstanceSpec{
+							ClusterServicePlanRef: &servicecatalogv1beta1.ClusterObjectReference{
+								Name: planName,
+							},
+							ClusterServiceClassRef: &servicecatalogv1beta1.ClusterObjectReference{
+								Name: serviceName,
+							},
+							Parameters: rawParams,
+						},
+					})
+				if err != nil {
+					return err
+				}
+
+				describe.ServiceInstance(cmd.OutOrStdout(), created)
+				return nil
+			}
+
+			namespacePlans, err := client.ServicecatalogV1beta1().
+				ServicePlans(p.Namespace).
+				List(metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+
+			for _, plan := range namespacePlans.Items {
+				if planName != plan.Spec.ExternalName {
+					continue
+				}
+				if serviceName != plan.Spec.ServiceClassRef.Name {
+					continue
+				}
+				if broker != "" && broker != plan.Spec.ServiceBrokerName {
+					continue
+				}
+
+				// plan found
+				created, err := client.ServicecatalogV1beta1().
+					ServiceInstances(p.Namespace).
+					Create(&servicecatalogv1beta1.ServiceInstance{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      instanceName,
+							Namespace: p.Namespace,
+						},
+						Spec: servicecatalogv1beta1.ServiceInstanceSpec{
+							ServicePlanRef: &servicecatalogv1beta1.LocalObjectReference{
+								Name: planName,
+							},
+							ServiceClassRef: &servicecatalogv1beta1.LocalObjectReference{
+								Name: serviceName,
+							},
+							Parameters: rawParams,
+						},
+					})
+				if err != nil {
+					return err
+				}
+
+				describe.ServiceInstance(cmd.OutOrStdout(), created)
+				return nil
+			}
+
+			return errors.New("no plan found")
```

Comment:
I think this should explicitly say which plan and service you were trying to use that way if you transpose the options the command kind of tells you that you got it wrong.